sky/backends/backend_utils.py

```diff
@@ -655,6 +664,27 @@ def _remove_multinode_config(
                 break
 
 
+def _restore_original_block_for_yaml(new_yaml: str, old_yaml: str,
+                                     key_names: Set[str]) -> str:
+    """Restore the original block with key_name recursively."""
```

Comment:
nit: Expand on 'recursively' a bit?  

```
Replaces 'new' with 'old' for all keys in key_names.

This is done recursively:  ..
```
actually no clear description comes to me.  This func is worth an example.

Reply:
Good point! I added a comment:
```
Replaces 'new' with 'old' for all keys in key_names.

The replacement will be applied recursively and only for the blocks
with the key in key_names, and have the same ancestors in both 'new'
and 'old' YAML tree.
```

I found the example will be a bit difficult to cover all the cases. Do you think the description is clear enough?

Reply:
Sgtm.